src/editions/SingleBatchEditionFactory.sol

```diff
@@ -141,15 +140,19 @@ contract SingleBatchEditionFactory {
         );
     }
 
-    function validateAttestation(SignedAttestation calldata signedAttestation) internal returns (bool) {
+    function validateAttestation(SignedAttestation calldata signedAttestation, address editionAddress)
+        public
+        view
+        returns (bool)
+    {
         // verify that the context for this attestation is valid
         address context = signedAttestation.attestation.context;
-        if (context != address(this)) {
-            revert UnexpectedContext(context);
+        if (context != editionAddress) {
+            revert AddressMismatch({expected: editionAddress, actual: context});
         }
 
-        // verify attestation
-        if (!showtimeVerifier.verifyAndBurn(signedAttestation)) {
+        // verify attestation without burning, because the editionAddress can not be reused
+        if (!showtimeVerifier.verify(signedAttestation)) {
```

Comment:
So here we don't need a burn mechanism anymore because the `editionAddress` we are predicting has to match what the backend has provided.
How the backend generate that address? With `editionId` and address of this contract like here: `ClonesUpgradeable.predictDeterministicAddress(editionImpl, bytes32(editionId), address(this))`?

And this is secure because only the backend has `editionId` and therefore can call `createEdition` with a valid `signedAttestation`.

One could re-call `createEdition` with a valid `signedAttestation` of an already deployed `editionId`, but that transaction would fail because contract would be trying to deploy at an address that is already used. Correct?

Reply:
there are 2 convenience methods to predict the address for an edition:

```solidity
    function getEditionId(EditionData calldata data, address creator) public pure returns (uint256 editionId);
    function getEditionAtId(uint256 editionId) public view returns (ISingleBatchEdition);
```

Both are pure/view functions so can be invoked for free via an RPC call, used to craft the attestation, sign it, and then relay the `createEdition` call. 

> One could re-call createEdition with a valid signedAttestation of an already deployed editionId, but that transaction would fail because contract would be trying to deploy at an address that is already used. Correct?

that's right! See [this test](https://github.com/showtime-xyz/showtime-contracts/blob/factory-optimize-attestation/test/editions/SingleBatchEditionTest.sol#L169-L181)